src/app/forecast/components/ui/PollenDetailsChart.tsx

```diff
@@ -1,77 +1,84 @@
-"use client";
+'use client';
 
-import { useEffect, useState } from "react";
-import { BiX } from "react-icons/bi";
+import { useEffect, useRef, useState } from 'react';
+import { BiX } from 'react-icons/bi';
 import {
   CartesianGrid,
-  Line,
   LineChart,
   ResponsiveContainer,
-  Tooltip,
   XAxis,
   YAxis,
-} from "recharts";
-import { LoadingSpinner } from "@/app/forecast/components";
+  Line,
+} from 'recharts';
+import { LoadingSpinner } from '@/app/forecast/components';
 import {
   usePartialLoadingStore,
-  useSearchLocationStore,
-} from "@/app/forecast/stores";
-interface PollenChartProps {
-  onClose?: () => void;
-}
+  usePollenDetailsChartStore,
+} from '@/app/forecast/stores';
+import { getPollenByApiKey, PollenApiKey } from '@/app/forecast/constants';
```

Comment:
import { getPollenByApiKey, PollenApiKey } from '@/app/forecast/constants';
```suggestion
import { getPollenByApiKey, type PollenApiKey } from '@/app/forecast/constants';
```